homeassistant/components/vicare/number.py

```diff
@@ -0,0 +1,194 @@
+    for description in sensor_descriptions:
+        for current in iterables:
+            suffix = ""
+            if len(iterables) > 1:
+                suffix = f" {current.id}"
+            entity = await hass.async_add_executor_job(
+                _build_entity,
+                f"{description.name}{suffix}",
+                current,
+                hass.data[DOMAIN][config_entry.entry_id][VICARE_DEVICE_CONFIG],
+                description,
+                hass,
+            )
+            if entity is not None:
+                entities.append(entity)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create the ViCare sensor devices."""
+    api = hass.data[DOMAIN][config_entry.entry_id][VICARE_API]
+
+    entities: list[ViCareNumber] = []
+    try:
+        await _entities_from_descriptions(
+            hass, entities, CIRCUIT_SENSORS, api.circuits, config_entry
+        )
+    except PyViCareNotSupportedFeatureError:
+        _LOGGER.debug("No circuits found")
+
+    async_add_entities(entities)
+
+
+class ViCareNumber(ViCareEntity, NumberEntity):
+    """Representation of a ViCare sensor."""
+
+    entity_description: ViCareNumberEntityDescription
+
+    def __init__(
+        self,
+        name: str,
+        api: PyViCareHeatingDeviceWithComponent,
+        device_config: PyViCareDeviceConfig,
+        description: ViCareNumberEntityDescription,
+        hass: HomeAssistant,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(device_config)
+        self.entity_description = description
+        self._attr_name = name
+        self._api = api
+        self._device_config = device_config
+        self._hass = hass
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._attr_native_value is not None
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID for this device."""
+        tmp_id = (
+            f"{self._device_config.getConfig().serial}-{self.entity_description.key}"
+        )
+        if hasattr(self._api, "id"):
+            return f"{tmp_id}-{self._api.id}"
+        return tmp_id
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set new value."""
+        if self.entity_description.value_setter:
+            await self._hass.async_add_executor_job(
+                self.entity_description.value_setter, self._api, value
+            )
+        self.async_write_ha_state()
+
+    def update(self) -> None:
+        """Update state of sensor."""
+        try:
+            with suppress(PyViCareNotSupportedFeatureError):
+                self._attr_native_value = self.entity_description.value_getter(
+                    self._api
+                )
+        except requests.exceptions.ConnectionError:
```

Comment:
```suggestion
        except ConnectionError:
```

just a personal preference.